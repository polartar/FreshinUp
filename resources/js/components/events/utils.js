import get from 'lodash/get'

const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}

export default getFileNameCopy
