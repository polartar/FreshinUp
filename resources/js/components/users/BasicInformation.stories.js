import { storiesOf } from '@storybook/vue'
import { action } from '@storybook/addon-actions'

import { FIXTURE_USER } from '../../../../tests/Javascript/__data__/user'

import BasicInformation from './BasicInformation.vue'
import { FIXTURE_USER_LEVELS } from '../../../../tests/Javascript/__data__/userLevels'
import { FIXTURE_USER_TYPES } from '../../../../tests/Javascript/__data__/userTypes'

import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'
import { FIXTURE_USER_STATUSES } from '../../../../tests/Javascript/__data__/userStatuses'
import { FIXTURE_USERS } from '../../../../tests/Javascript/__data__/users'
import { FIXTURE_COMPANIES } from '../../../../tests/Javascript/__data__/companies'

const mock = new MockAdapter(axios)

mock.onGet(/users/)
  .reply(200, {
    data: FIXTURE_USERS
  })

mock.onGet(/companies/)
  .reply(200, {
    data: FIXTURE_COMPANIES
  })

export const Default = () => ({
  components: { BasicInformation },
  template: `
    <v-container>
      <basic-information/>
    </v-container>
  `
})

export const IsLoading = () => ({
  components: { BasicInformation },
  template: `
    <v-container>
      <basic-information is-loading/>
    </v-container>
  `
})

const methods = {
  onSave (payload) {
    action('onSave')(payload)
  },
  onCancel () {
    action('onCancel')()
  },
  onDelete (payload) {
    action('onDelete')(payload)
  },
  onChangePassword (payload) {
    action('onChangePassword')(payload)
  }
}

export const BasicView = () => ({
  components: { BasicInformation },
  data () {
    return {
      user: FIXTURE_USER,
      levels: FIXTURE_USER_LEVELS,
      types: FIXTURE_USER_TYPES,
      statuses: FIXTURE_USER_STATUSES
    }
  },
  methods,
  template: `
    <v-container>
      <basic-information
        :value="user"
        :levels="levels"
        :types="types"
        :statuses="statuses"
        @input="onSave"
        @cancel="onCancel"
        @delete="onDelete"
        @change-password="onChangePassword"
      />
    </v-container>
  `
})

export const AdminView = () => ({
  components: { BasicInformation },
  data () {
    return {
      user: FIXTURE_USER,
      levels: FIXTURE_USER_LEVELS,
      types: FIXTURE_USER_TYPES,
      statuses: FIXTURE_USER_STATUSES
    }
  },
  methods,
  template: `
    <v-container>
      <basic-information
        :value="user"
        :levels="levels"
        :types="types"
        :statuses="statuses"
        is-admin
        @input="onSave"
        @cancel="onCancel"
        @delete="onDelete"
        @change-password="onChangePassword"
      />
    </v-container>
  `
})

storiesOf('FoodFleet|components/users/BasicInformation', module)
  .addParameters({
    backgrounds: [
      { name: 'default', value: '#f1f3f6', default: true }
    ]
  })
  .add('Default', Default)
  .add('IsLoading', IsLoading)
  .add('BasicView', BasicView)
  .add('AdminView', AdminView)
