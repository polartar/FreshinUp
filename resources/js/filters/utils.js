export function prefixStr (value, prefix = '@') {
  if (!value) return ''
  return value ? `${prefix} ${value}` : ''
}
