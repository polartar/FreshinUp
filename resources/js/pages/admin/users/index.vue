<template>
  <div>
    <v-flex
      d-inline-flex
      align-center
      ma-2
    >
      <h2 class="white--text">
        {{ pageTitle }}
      </h2>
      <v-btn
        slot="activator"
        color="primary"
        dark
        class="ml-4"
        @click="userNew"
      >
        New User
      </v-btn>
    </v-flex>
    <v-divider />

    <br>
    <user-filter
      v-if="!isLoading"
      :sortables="sortables"
      @runFilter="filterUsers"
    />
    <user-list
      v-if="!isLoading"
      :users="users"
      :levels="userlevels"
      :statuses="userstatuses"
      :is-loading="isLoading || isLoadingList"
      :rows-per-page="pagination.rowsPerPage"
      :page="pagination.page"
      :total-items="pagination.totalItems"
      :sort-by="sorting.sortBy"
      :descending="sorting.descending"
      :headers="headers"
      :item-actions="itemActions"
      class="users-list"
      must-sort
      @paginate="onPaginate"
      @change-status="changeStatus"
      @change-level="changeLevel"
      @manage-view="userView"
      @manage-edit="userEdit"
      @manage-delete="deleteUser"
      @manage-multiple-delete="deleteMultiple"
    >
      <template v-slot:item-inner-name,email="{ item }">
        <div class="subheading">
          {{ item.name }}
        </div>
        {{ item.email }}
      </template>
      <template v-slot:item-inner-user_type="{ item }">
        <template v-if="item.user_type">
          {{ item.user_type.name }}
        </template>
      </template>
      <template v-slot:item-inner-company_name="{ item }">
        {{ item.company_name }}
      </template>
    </user-list>

    <v-dialog
      v-model="deleteUserDialog"
      max-width="500"
    >
      <simple-confirm
        :class="{ 'deleting': deletablesProcessing }"
        :title="deleteDialogTitle"
        ok-label="Yes"
        cancel-label="No"
        @ok="deleteUsers"
        @cancel="deleteUserDialog = false"
      >
        <div class="py-5 px-2">
          <template v-if="deletablesProcessing">
            <div class="text-xs-center">
              <p class="subheading">
                Processing, please wait...
              </p>
              <v-progress-circular
                :rotate="-90"
                :size="200"
                :width="15"
                :value="deletablesProgress"
                color="primary"
              >
                {{ deletablesStatus }}
              </v-progress-circular>
            </div>
          </template>
          <template v-else>
            <p class="subheading">
              <span v-if="deletables.length < 2">User</span>
              <span v-else> Users</span>
              : {{ deleteableNames }}
            </p>
          </template>
        </div>
      </simple-confirm>
    </v-dialog>
  </div>
</template>

<script>
import { mapGetters, mapActions, mapState } from 'vuex'
import UserList from 'fresh-bus/components/datatable/user-list.vue'
import userFilter from 'fresh-bus/components/users/FilterSorter.vue'
import simpleConfirm from 'fresh-bus/components/SimpleConfirm.vue'
import { deletables } from 'fresh-bus/components/mixins/Deletables'
import get from 'lodash/get'

export default {
  layout: 'admin',
  components: {
    UserList,
    userFilter,
    simpleConfirm
  },
  mixins: [deletables],
  data () {
    return {
      pageTitle: 'Users',
      deleteUserDialog: false,
      user: {},
      lastFilterParams: {},
      headers: [
        { text: 'Status', sortable: true, value: 'status', align: 'center' },
        { text: 'Name / Email', value: 'name,email', align: 'left' },
        { text: 'User Type', sortable: true, value: 'user_type', align: 'center' },
        { text: 'Company', value: 'company_name', align: 'left' },
        { text: 'BUS Role', sortable: true, value: 'level', align: 'center' },
        { text: 'Manage', sortable: false, value: 'manage', align: 'center' }
      ],
      itemActions: [
        { action: 'view', text: 'View' },
        { action: 'edit', text: 'Edit' },
        { action: 'delete', text: 'Delete' }
      ]
    }
  },
  computed: {
    isLoadingList () {
      return get(this.$store, 'state.users.pending.items', true)
    },
    ...mapGetters('users', {
      users: 'items',
      pagination: 'pagination',
      sorting: 'sorting',
      sortBy: 'sortBy'
    }),
    ...mapGetters('page', ['isLoading']),
    ...mapGetters('userLevels', { 'userlevels': 'items' }),
    ...mapGetters('userTypes', { 'userTypes': 'items' }),
    ...mapGetters('userStatuses', { 'userstatuses': 'items' }),
    ...mapState('users', ['sortables']),
    deleteDialogTitle () {
      return this.deletables.length < 2 ? 'Are you sure you want to delete this user?' : 'Are you sure you want to delete the following users?'
    },
    deleteableNames () {
      return this.deletables.map((user) => {
        return user.name
      }).join(', ')
    }
  },
  methods: {
    ...mapActions('page', {
      setPageLoading: 'setLoading'
    }),

    changeStatus (status, user) {
      this.$store.dispatch('users/patchItem', { data: { status }, params: { id: user.id } }).then(() => {
        this.filterUsers(this.lastFilterParams)
      })
    },
    changeLevel (level, user) {
      this.$store.dispatch('users/patchItem', { data: { level }, params: { id: user.id } }).then(() => {
        this.filterUsers(this.lastFilterParams)
      })
    },
    userNew () {
      this.$router.push({ path: '/admin/users/new' })
    },
    userView (user) {
      this.$router.push({ path: '/admin/users/' + user.id })
    },
    userEdit (user) {
      this.$router.push({ path: '/admin/users/' + user.id + '/edit' })
    },
    deleteUserDialogUp (user) {
      this.deleteUserDialog = true
      this.user = user
    },
    deleteUser (user) {
      this.deleteDialogUp(user)
    },
    async deleteUsers () {
      this.deletablesProcessing = true
      this.deletablesProgress = 0
      this.deletablesStatus = ''
      let dispatcheables = []

      this.deletables.forEach((user) => {
        dispatcheables.push(this.$store.dispatch('users/deleteItem', { getItems: false, params: { id: user.id } }))
      })

      let chunks = this.chunk(dispatcheables, this.deletablesParrallelRequest)
      let doneCount = 0

      for (let i in chunks) {
        await Promise.all(chunks[i])
        doneCount += chunks[i].length
        this.deletablesStatus = doneCount + ' / ' + this.deletables.length + ' Done'
        this.deletablesProgress = doneCount / this.deletables.length * 100
        await this.sleep(this.deletablesSleepTime)
      }

      this.filterUsers(this.lastFilterParams)
      await this.sleep(500)
      this.deleteUserDialog = false
      this.deletablesProcessing = false
    },
    deleteMultiple (users) {
      this.deleteDialogUp(users)
    },
    deleteDialogUp (users) {
      if (!Array.isArray(users)) {
        users = [users]
      }
      this.deleteUserDialog = true
      this.deletables = users
    },
    onPaginate (value) {
      this.$store.dispatch('users/setPagination', value)
      this.$store.dispatch('users/getItems', { params: { include: ['user_type'] } })
    },
    filterUsers (params) {
      this.lastFilterParams = params
      this.$store.dispatch('users/setSort', params.sort)
      this.$store.dispatch('users/setFilters', {
        ...this.$route.query,
        ...this.lastFilterParams
      })
      this.$store.dispatch('users/getItems', { params: { include: ['user_type'] } })
    }
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.setPageLoading(true)
    vm.$store.dispatch('users/setFilters', {
      ...vm.$route.query
    })
    Promise.all([
      vm.$store.dispatch('userLevels/getUserlevels'),
      vm.$store.dispatch('userTypes/getItems'),
      vm.$store.dispatch('userStatuses/getUserstatuses')
    ]).then(() => {
      vm.$store.dispatch('page/setLoading', false)
      if (next) next()
    })
      .catch((error) => { console.error(error) })
      .then(() => {
        vm.$store.dispatch('page/setLoading', false)
      })
  }
}
</script>
