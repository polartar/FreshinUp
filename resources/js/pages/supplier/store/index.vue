
<script>
import stores from '~/pages/admin/fleet-members/index.vue'
import { mapGetters } from 'vuex'
const INCLUDE = [
  'tags',
  'addresses',
  'status',
  'owner',
  'type'
]

export default {
  extends: stores,
  computed: {
    ...mapGetters(['currentUser'])
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('stores/setFilters', {
        ...vm.$route.query,
        ...this.lastFilterParams,
        owner_uuid: vm.currentUser.uuid
      }),
      vm.$store.dispatch('storeStatuses/getItems', {
        params: { include: INCLUDE }
      })
    ])
      .then(() => {
        if (next) next()
      })
      .catch((error) => console.error(error))
      .then(() => vm.setPageLoading(false))
  }
}
</script>
