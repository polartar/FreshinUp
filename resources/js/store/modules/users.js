import busUsers from 'fresh-bus/store/modules/users'
import axios from 'axios'

export default ({ items, item }) => {
  const store = busUsers({ items, item }) //
  store.actions = {
    ...store.actions,
    createCustomerOrSupplier (context, { data }) {
      return new Promise((resolve, reject) => {
        axios({
          url: '/foodfleet/users/customer-or-supplier',
          method: 'POST',
          data
        })
          .then(response => resolve(response.data))
          .catch(error => reject(error))
      })
    }
  }
  return {
    ...store,
    namespaced: true
  }
}
