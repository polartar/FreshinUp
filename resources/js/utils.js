import get from 'lodash/get'
/**
 *
 * @param options HTML attributes
 * @param [options.type] string 'link'|'script' or anything in fact
 * @param [options.container] HTMLElement
 * @returns {Promise<unknown>}
 */
export const lazyLoad = (options) => {
  return new Promise((resolve, reject) => {
    if (!options.type) {
      return reject(new Error('options.type is required'))
    }
    if (!options.container) {
      options.container = document.body
    }
    const { type, container, ...attributes } = options
    let element = document.createElement(type)
    Object.keys(attributes).forEach(key => element.setAttribute(key, attributes[key]))
    element.addEventListener('load', () => {
      resolve(element)
    })
    element.addEventListener('error', (e) => {
      reject(e)
    })
    container.append(element)
  })
}

export const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}
