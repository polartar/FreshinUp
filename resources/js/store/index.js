import { createStore } from 'fresh-bus/store'

import page from './modules/page'
import devices from './modules/devices'
import financialModifiers from './modules/financialModifiers'
import financialReports from './modules/financialReports'
import paymentTypes from './modules/paymentTypes'
import financialsummary from './modules/financialsummary'
import squares from './modules/squares'
import documents from './modules/documents'
import documentStatuses from './modules/documentStatuses'
import documentTypes from './modules/documentTypes'
import transactions from './modules/transactions'
import companyDetails from './modules/companyDetails'
import eventSummary from './modules/eventSummary'
import eventTypes from './modules/eventTypes'
import eventHistories from './modules/eventHistories'
import venues from './modules/venues'
import stores from './modules/stores'
import storeStatuses from './modules/storeStatuses'
import storeTypes from './modules/storeTypes'
import companies from './modules/companies'

export default (initialState = {}) => {
  return createStore(
    initialState,
    {
      modules: {
        page: page({}),
        devices: devices({}),
        financialModifiers: financialModifiers({}),
        financialReports: financialReports({}),
        paymentTypes: paymentTypes({}),
        financialsummary: financialsummary({}),
        squares: squares({}),
        documents: documents({}),
        documentStatuses: documentStatuses({}),
        documentTypes: documentTypes({}),
        transactions: transactions({}),
        companyDetails: companyDetails({}),
        eventSummary: eventSummary({}),
        eventTypes: eventTypes({}),
        eventHistories: eventHistories({}),
        venues: venues({}),
        stores: stores({}),
        storeStatuses: storeStatuses({}),
        storeTypes: storeTypes({}),
        companies: companies({})
      }
    }
  )
}
