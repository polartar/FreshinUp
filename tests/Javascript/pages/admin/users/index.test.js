import { shallowMount } from '@vue/test-utils'
import { createStore } from 'fresh-bus/store'
import { createLocalVue } from 'fresh-bus/tests/utils'
import { FIXTURE_USERS_RESPONSE, FIXTURE_USERS_SORTED_BY_FIRSTNAME } from 'tests/__data__/users'
import { FIXTURE_CURRENT_USER } from 'tests/__data__/user'
import Component from '~/pages/admin/users/index.vue'

describe('Admin Users Page', () => {
  let localVue, mock
  describe('Mount', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
      mock = vue.mock
    })
    afterEach(() => {
      mock.restore()
    })
    test('snapshot', async () => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
      mock = vue.mock
        .onGet('api/currentUser').reply(200, FIXTURE_CURRENT_USER)
        .onGet('api/users', { params: { 'page[size]': 10, 'page[number]': 1, sort: 'first_name' } })
        .reply(200, { data: FIXTURE_USERS_SORTED_BY_FIRSTNAME })
        .onGet('api/users').reply(200, FIXTURE_USERS_RESPONSE)
        .onAny()
        .reply(config => {
          console.warn('No mock match for ' + config.url, config)
          return [404, {}]
        })
      const store = createStore({
        users: {
          items: FIXTURE_USERS_RESPONSE // Start with listed sorted by id
        }
      })
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })
      // Action: change State Machine's state
      await wrapper.vm.$store.dispatch('page/setLoading', false)
      await wrapper.vm.$nextTick()
      await wrapper.vm.$store.dispatch('users/getItems')
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
