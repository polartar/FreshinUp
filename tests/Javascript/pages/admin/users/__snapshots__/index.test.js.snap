// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Users Page Mount snapshot 1`] = `
<div>
  <v-flex-stub
    align-center=""
    d-inline-flex=""
    ma-2=""
    tag="div"
  >
    <h2>
      Users
    </h2>
     
    <v-btn-stub
      activeclass="v-btn--active"
      class="ml-4"
      color="primary"
      dark="true"
      slot="activator"
      tag="button"
      type="button"
    >
      
      New User
    
    </v-btn-stub>
  </v-flex-stub>
   
  <v-divider-stub />
   
  <br />
   
  <user-filter-stub
    autocompleteurl="/api/users"
    companies=""
    sortables="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
  />
   
  <user-list-stub
    class="users-list"
    headers="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
    itemactions="[object Object],[object Object],[object Object]"
    levels=""
    must-sort=""
    page="1"
    rowsperpage="10"
    rowsperpageitems="5,10,15,25,30,50"
    sortby=""
    statuses=""
    totalitems="79"
    users="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
  />
   
  <v-dialog-stub
    contentclass=""
    maxwidth="500"
    origin="center center"
    transition="dialog-transition"
    width="auto"
  >
    <simple-confirm-stub
      cancellabel="No"
      class=""
      oklabel="Yes"
      title="Are you sure you want to delete this user?"
    >
      <div
        class="py-5 px-2"
      >
        <p
          class="subheading"
        >
          <span>
            User
          </span>
          
            : 
          
        </p>
      </div>
    </simple-confirm-stub>
  </v-dialog-stub>
</div>
`;
